source/common/tracing/zipkin/zipkin_core_types.cc

```diff
@@ -71,6 +71,14 @@ Annotation& Annotation::operator=(const Annotation& ann) {
   return *this;
 }
 
+void Annotation::changeEndpointServiceName(const std::string& service_name) {
+  if (endpoint_.valid()) {
+    Endpoint endpoint(endpoint_.value());
```

Comment:
I see, sorry, yes. I'm not sure off the top of my head the right way to deal with this. I would just copy the code and then make sure you have test coverage for both variants in optional_test (just declare a const Optional for testing).